### ISSUE TITLE
Apply for a license.

### ISSUE THREAD

**Issue by liuruoze** [bmnielsen/Locutus#2]
Hello, Nielsen,

I am a Ph.D. student at Nanjing University in China. Our lab intends to form a team to participate in the COG StarCraft AI competition in July and the AIIDE StarCraft AI competition in October. We intend to develop a bot on Locutus, especially based on the variant of Locutus which is CSE.

We plan to improve on the following sections:
1. At the code level, we intend to implement the micro-action code of the High Templar and the Arbiter. We want to make effective use of magic, such as the Psionic Storm and Hallucination of the High Templar, and the Recall and Stasis Field of the Arbiter. We hope that magic will make the bot perform better when fighting with Terran and Zerg.
2. CSE uses machine learning techniques for forecasting. We will implement a new machine learning module which may use more advanced algorithms such as LSTM. Through this, we hope to achieve more accurate predictions.
3. We will improve some of the micro-action modules. For example, improve the placement of the defensive buildings to better resist the enemy's attack.
4. More tactics of human player will be introduced, such as airlift more unit types (not only Dark Templar) to conduct sneak attacks.

We don't know if these improvements can be considered a big enough change. We hope to get permission from your agreement so that we can participate in the following StarCraft AI competition. 

Sincerely look forward to your reply.

**Comment by bmnielsen** [comment]
Hello,

If you are basing the bot on CSE, then you do not need my permission. Both CSE and the AIIDE 2018 version of Locutus were released under the MIT license, so you are free to use either of them.

You would need my permission to use any of the changes that I have released since AIIDE 2018, but based on your list of ideas, I don't think any of the changes would be of much use to you anyway (I haven't had much time to work on my bot recently, and the changes I have made were mainly to combat specific opponents in the SSCAIT tournament).

Given the above, and the fact that I want to be conservative about granting permission for forks that play Protoss, I am going to decline to give permission to base your bot on the current version on GitHub. But I encourage you to use either the CSE or Locutus AIIDE 2018 bots as a starting point, which should be approximately equivalent anyway.

Good luck!

Bruce
